# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmBCClass.h
# Commit: ce799cb566ac
# Full Hash: ce799cb566ac0fb95468118608132be11db6a359
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2022-03-24 21:54:20
# Regressor Bug: 1642412
# File Overlap Count: 19
# Description:
#   Bug 1642412 - wasm: Rollup of inline table.get/set for anyref. r=lth
#   
#   Rollup of the previous inline table.get/set patches.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D141024
# ==============================================================================

diff -r c10609d434ae -r ce799cb566ac js/src/wasm/WasmBCClass.h
--- a/js/src/wasm/WasmBCClass.h	Thu Mar 24 17:40:32 2022 +0000
+++ b/js/src/wasm/WasmBCClass.h	Thu Mar 24 17:56:50 2022 +0000
@@ -1080,6 +1080,15 @@
 
   //////////////////////////////////////////////////////////////////////
   //
+  // Table access.
+
+  Address addressOfTableField(const TableDesc& table, uint32_t fieldOffset,
+                              RegPtr tls);
+  void loadTableLength(const TableDesc& table, RegPtr tls, RegI32 length);
+  void loadTableElements(const TableDesc& table, RegPtr tls, RegPtr elements);
+
+  //////////////////////////////////////////////////////////////////////
+  //
   // Heap access.
 
   void bceCheckLocal(MemoryAccessDesc* access, AccessCheck* check,
@@ -1542,6 +1551,10 @@
   [[nodiscard]] bool emitTableSet();
   [[nodiscard]] bool emitTableSize();
 
+  void emitTableBoundsCheck(const TableDesc& table, RegI32 index, RegPtr tls);
+  [[nodiscard]] bool emitTableGetAnyRef(uint32_t tableIndex);
+  [[nodiscard]] bool emitTableSetAnyRef(uint32_t tableIndex);
+
 #ifdef ENABLE_WASM_GC
   [[nodiscard]] bool emitStructNewWithRtt();
   [[nodiscard]] bool emitStructNewDefaultWithRtt();